Docs/Resource Manager.md

```diff
@@ -0,0 +1,308 @@
+## Resource Manager
+
+### Functional Overview
+
+The Resource Manager processes requests for assets,
+finds those assets in memory, disk cache, or from network servers,
+and notifies the requestor when that asset is available.
+It is also responsible for notifying users of an asset when that asset is no longer available.
```

Comment:
What's the deal with the odd newline placement?

Reply:
For freetext documentation like this, I try to put the line breaks at the end of each sentence (and after commas if the sentence is long). Compared to always wrapping at a fixed column, this works better with version control - changing a few words doesn't require you to rewrap the entire remaining paragraph (which leads to unnecessary changes in merges and `git blame`). In the rendered output, it looks the same either way, because single newlines are ignored by Markdown.

But if you prefer, I can rewrap everything to a fixed column width. (How many columns do we use here usually?)

Reply:
I think you have a good point. I'm not sure what the answer is here. README.md seems to break at 80 columns while CONTRIBUTING.md doesn't break, ever. Any thoughts @Deledrius, @dpogue?

Reply:
80 columns is really nice for reading in the terminal, and &mdash; as @dgelessus mentioned &mdash; is absolutely terrible for rewording or editing anything because you have to redo all the line breaks.
These docs are mostly historical and unlikely to change, so maybe that's fine?

Having each paragraph on its own line is the other alternative I usually tend towards. Most reasonable editors will wrap it, so you can just size your editor to how wide you want it to be for comfortable reading.

Reply:
Based on the discussion, I think this would be better if the line breaks were changed to occur around 80 columns. Thanks for converting these documents to markdown for us!